packages/react-components/react-timepicker-compat-preview/src/components/TimePicker/TimePicker.types.ts

```diff
@@ -1,17 +1,99 @@
-import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
+import type { ComboboxSlots, ComboboxState, ComboboxProps, SelectionEvents } from '@fluentui/react-combobox';
 
-export type TimePickerSlots = {
-  root: Slot<'div'>;
+export type Hour =
+  | 0
+  | 1
+  | 2
+  | 3
+  | 4
+  | 5
+  | 6
+  | 7
+  | 8
+  | 9
+  | 10
+  | 11
+  | 12
+  | 13
+  | 14
+  | 15
+  | 16
+  | 17
+  | 18
+  | 19
+  | 20
+  | 21
+  | 22
+  | 23
+  | 24;
+
+export type TimePickerSlots = ComboboxSlots;
+
+export type TimeSelectionEvents = SelectionEvents;
+export type TimeSelectionData = {
+  selectedTime: Date | undefined;
 };
 
 /**
  * TimePicker Props
  */
-export type TimePickerProps = ComponentProps<TimePickerSlots> & {};
+export type TimePickerProps = Omit<
+  ComboboxProps,
+  // Omit children as TimePicker has predefined children
+  | 'children'
```

Comment:
**discussion (doesn't block)**

I see that you've omitted children. Do you have a plan for how users would be able to customize each of the TimePicker options? 

v8 has a `formateDateString` prop https://developer.microsoft.com/en-us/fluentui#/controls/web/timepicker.

Reply:
Yep that's in the plan. V8 has this prop for formatting date string in dropdown: https://github.com/microsoft/fluentui/blob/b7dbe130771cb7941f6bd3d00a81deab0ca30ef2/packages/react/src/components/TimePicker/TimePicker.types.ts#L108C41-L108C41
I'll add something similar.